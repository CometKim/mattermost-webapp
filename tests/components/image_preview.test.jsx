import React from 'react';

import {shallow} from 'enzyme';

import ImagePreview from 'components/image_preview';

describe('components/ImagePreview', () => {
    test('should match snapshot', () => {
        const fileInfo = {
            id: '{file_id}',
        };

        const makeWrapper = (fileInfo) => shallow(
            ImagePreview({fileInfo})
        );

        expect(makeWrapper(fileInfo)).toMatchSnapshot();

        fileInfo.has_preview_image = true;
        expect(makeWrapper(fileInfo)).toMatchSnapshot();
    });
});
