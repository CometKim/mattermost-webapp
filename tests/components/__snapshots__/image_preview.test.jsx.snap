// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/ImagePreview should match snapshot 1`] = `
<a
  download={true}
  href="/api/v4/files/{file_id}"
  rel="noopener noreferrer"
  target="_blank"
>
  <img
    src="/api/v4/files/{file_id}"
  />
</a>
`;

exports[`components/ImagePreview should match snapshot 2`] = `
<a
  download={true}
  href="/api/v4/files/{file_id}"
  rel="noopener noreferrer"
  target="_blank"
>
  <img
    src="/api/v4/files/{file_id}/preview"
  />
</a>
`;
